app/main.py

```diff
@@ -1,3 +1,44 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    caching_result = {}
+    wraps(func)
+
+    def wrapper(*args: tuple) -> Any:
```

Comment:
There is no need to use type annotation for `args` and `kwargs`, unless you need to specify their content. You can omit type hinting for them.